stl_to_obj/textures.py

```diff
@@ -0,0 +1,57 @@
+import json
+
+# filenames of all the PBR textures included in game in the textures folder, stored as a dict for each file
+TEXTURE_FOLDER = "textures"
+
+NORMAL_OPTIONS = "normal_enabled = true\nnormal_scale = -1.0\n"
+DEPTH_OPTIONS = "depth_enabled = true\ndepth_scale = -0.01\ndepth_deep_parallax = false\ndepth_flip_tangent = false\ndepth_flip_binormal = false\n"
+
+class Texture:
+    def __init__(self, folder, jpg_dict, uv1_scale = ""):
+        self.folder = folder
+        self.jpg_dict = jpg_dict
+        # uv shouldn't be in the dict because it doesn't need an ext_resource like the .jpgs
+        self.uv1_scale = uv1_scale 
+
+# dict of all the textures we already have
+texture_dict = dict()
+other_textures = dict()
+
+# add textures from a .json file to dict 
+# allows contributors to use their own textures (that we don't have)
+def add_textures(filename):
+    with open(filename, 'r') as f:
+        # load JSON object as dictionary
+        data = json.load(f)
+
+        textures = data["textures"]
+        
+        for t in textures:
+            # create Texture
+            folder = t["folder"]
+            jpg_dict = t["jpg_dict"]
+            uv1_scale = ""
+            if "uv1_scale" in t:
+                uv1_scale = t["uv1_scale"]
+            texture = Texture(
+                folder = folder,
+                jpg_dict = jpg_dict,
+                uv1_scale = uv1_scale
+            )
+
+            # add it to texture_dict
+            texture_dict[folder] = texture
+
+        # load in textures that don't require .jpgs
+        if "other_textures" in data:
+            for key in data["other_textures"]:
+                other_textures[key] = data["other_textures"][key]
+
+# debug printout to check if we load our provided textures correctly
+if __name__ == "__main__":
+    add_textures("textures.json")
+    for i in texture_dict:
+        print(i)
+
+    for i in other_textures:
+        print(i)
```

Comment:
Add an option to your `argparse` to list the textures and otherwise don't print any output.

Reply:
I don't think this is necessary, because usually this module is used by other python scripts like `stl_to_tscn.py` instead of being run itself. Adding the argparse allows us to hide the debug printout, but when we run this module itself and we don't see any debug printout, it appears like the module did nothing. 

Maybe I should just remove the `if __name__ == "__main__":` section then?